### PR TITLE
Added View Basic tests for accessibilityLabel and importantForAccessibility

### PR DESCRIPTION
**Pull request by zholobov**
Added View Basic tests for accessibilityLabel recursive generation and importantForAccessibility.

### CODE REVIEW

File: samples/RXPTest/src/Tests/ViewBasicTest.tsx [hunk 1]
```diff
@@ -106,6 +110,59 @@ class BasicView extends RX.Component<RX.CommonProps, RX.Stateless> {
...
+                            </RX.View>
+                        </RX.View>
+                        <RX.Text style={ _styles.labelText }>
+                            { 'Delta' }
+                        </RX.Text>
+                        <RX.Text 
+                            style={ _styles.labelText } 
+                            importantForAccessibility={ RX.Types.ImportantForAccessibility.No }
+                        >
+                            { 'Epsilon' }
```
Comment (rigdern) [inline samples/RXPTest/src/Tests/ViewBasicTest.tsx]: Should we verify the behavior of nesting something inside of a `Text` with `importantForAccessibility=No`?

File: samples/RXPTest/src/Tests/ViewBasicTest.tsx [hunk 2]
```diff
@@ -106,6 +110,59 @@ class BasicView extends RX.Component<RX.CommonProps, RX.Stateless> {
...
+                        { 'Should be read by screen reader as \"Alpha Beta Gamma Delta\" ' + 
+                          'and screen reader should not be able to get to individual texts.' }
+                    </RX.Text>
+                </RX.View>
+                <RX.View style={ _styles.labelContainer }>
+                    <RX.View 
+                        style={ _styles.labelContainer } 
+                        importantForAccessibility={ RX.Types.ImportantForAccessibility.Yes }
+                    >
+                        <RX.Text style={ _styles.labelText }>
```
Comment (rigdern) [inline samples/RXPTest/src/Tests/ViewBasicTest.tsx]: Should we verify the behavior of explicitly setting `importantForAccessibility` to `Auto`?

File: samples/RXPTest/src/Tests/ViewBasicTest.tsx [hunk 3]
```diff
@@ -106,6 +110,59 @@ class BasicView extends RX.Component<RX.CommonProps, RX.Stateless> {
                           'I have differing border radii' }
                     </RX.Text>
                 </RX.View>
+
+                <RX.View
```
Comment (rigdern) [inline samples/RXPTest/src/Tests/ViewBasicTest.tsx]: Should we note somewhere that the intention of this is to test `accessibilityLabel` and `importantForAccessibility` on `Text` components?

File: samples/RXPTest/src/Tests/ViewBasicTest.tsx (hunk 1 above)
Comment (gmangov) [reply samples/RXPTest/src/Tests/ViewBasicTest.tsx]: In the current implementation if you try two-level nesting inside RX.Text, you get an exception

File: samples/RXPTest/src/Tests/ViewBasicTest.tsx (hunk 3 above)
Comment (zholobov) [reply samples/RXPTest/src/Tests/ViewBasicTest.tsx]: Actually it's mostly about View, not about Text. I will add comment in code.

---
In reply to: [161926917](https://github.com/Microsoft/reactxp/pull/431#discussion_r161926917) [](ancestors = 161926917)

File: samples/RXPTest/src/Tests/ViewBasicTest.tsx (hunk 2 above)
Comment (zholobov) [reply samples/RXPTest/src/Tests/ViewBasicTest.tsx]: Good idea to test Auto too! Will do.

---
In reply to: [161925938](https://github.com/Microsoft/reactxp/pull/431#discussion_r161925938) [](ancestors = 161925938)

File: samples/RXPTest/src/Tests/ViewBasicTest.tsx (hunk 1 above)
Comment (zholobov) [reply samples/RXPTest/src/Tests/ViewBasicTest.tsx]: This does not seem to be nesting anything inside RX.Text. Can you elaborate?

---
In reply to: [161927956](https://github.com/Microsoft/reactxp/pull/431#discussion_r161927956) [](ancestors = 161927956)

Comment (zholobov) [reply samples/RXPTest/src/Tests/ViewBasicTest.tsx]: That would be Text tests, probably should go to TextTest.tsx - not part of this View test PR.

---
In reply to: [161925535](https://github.com/Microsoft/reactxp/pull/431#discussion_r161925535) [](ancestors = 161925535)

Comment (gmangov) [reply samples/RXPTest/src/Tests/ViewBasicTest.tsx]: There isn't. I am just replying to the proposal to test nesting inside Text. Two level nesting will cause exception in the present implementation, i.e.
<RX.Text>
   xxxx
   <RX.View>
    <RX.Text>
     yyyyyy

Comment (zholobov) [reply samples/RXPTest/src/Tests/ViewBasicTest.tsx]: Sorry, CF does not show your replies as part of thread. Can you use CF to review the PRs?

---
In reply to: [161929225](https://github.com/Microsoft/reactxp/pull/431#discussion_r161929225) [](ancestors = 161929225)

File: samples/RXPTest/src/Tests/ViewBasicTest.tsx (hunk 3 above)
Comment (gmangov) [reply samples/RXPTest/src/Tests/ViewBasicTest.tsx]: RX.Text implementation is still in the works. Maybe the tests for RX.Text specifically should be added once the implementation is in place....

**Comment by erictraut** [comment]
These changes look good to me. I can merge them if & when you're done incorporating any further feedback. Just let me know.

File: samples/RXPTest/src/Tests/ViewBasicTest.tsx (hunk 3 above)
Comment (zholobov) [reply samples/RXPTest/src/Tests/ViewBasicTest.tsx]: great catch! I'm removing incidental RX.Text tests from here and replacing them with explicit View tests.

---
In reply to: [161929548](https://github.com/Microsoft/reactxp/pull/431#discussion_r161929548) [](ancestors = 161929548)

**Comment by zholobov** [comment]
Added second commit addressing the feedback. @rigdern @gmangov could you please review and close comments that have been addressed?

File: samples/RXPTest/src/Tests/ViewBasicTest.tsx [hunk 4]
```diff
@@ -106,6 +110,87 @@ class BasicView extends RX.Component<RX.CommonProps, RX.Stateless> {
...
+                    </RX.Text>
+                </RX.View>
+                <RX.View style={ _styles.labelContainer }>
+                    { BasicView.accessibilityLabelAndImportantForAccessibilityTestUI(RX.Types.ImportantForAccessibility.No) }
+                </RX.View>
+            </RX.View>
+        );
+    }
+
+    private static accessibilityLabelAndImportantForAccessibilityTestUI(important: RX.Types.ImportantForAccessibility) {
```
Comment (rigdern) [inline samples/RXPTest/src/Tests/ViewBasicTest.tsx]: This could be a free-floating function in this file rather than a `static` method.

**Comment by zholobov** [comment]
@erictraut I addressed all feedback. Could you please merge this?

**Pull request merged**